### ISSUE TITLE
bug: inlay hints stopped working with latest neovim nightly

### ISSUE THREAD

**Issue by mati865** [LazyVim/LazyVim#2005]
### Did you check docs and existing issues?

- [X] I have read all the LazyVim docs
- [X] I have searched the existing issues of LazyVim
- [X] I have searched the existing issues of plugins related to this issue

### Neovim version (nvim -v)

NVIM v0.10.0-dev-748eae6

### Operating system/version

Ubuntu 22.04

### Describe the bug

I've been using inlay hints for a long time with Rust code but it broke in recent nightly and shows this stack trace when attaching to the file:
```
   Error  16:08:58 msg_show.lua_error Error detected while processing LspAttach Autocommands for "*":
16:08:58 msg_show Error executing lua callback: ...share/nvim/lazy/LazyVim/lua/lazyvim/plugins/lsp/init.lua:123: attempt to call upvalue 'inlay_hint' (a table value)
stack traceback:
	...share/nvim/lazy/LazyVim/lua/lazyvim/plugins/lsp/init.lua:123: in function 'on_attach'
	.../.local/share/nvim/lazy/LazyVim/lua/lazyvim/util/lsp.lua:32: in function <.../.local/share/nvim/lazy/LazyVim/lua/lazyvim/util/lsp.lua:29>
	[C]: in function 'nvim_exec_autocmds'
	.../nightly/nvim-linux64/share/nvim/runtime/lua/vim/lsp.lua:1722: in function '_on_attach'
	.../nightly/nvim-linux64/share/nvim/runtime/lua/vim/lsp.lua:1472: in function ''
	vim/_editor.lua: in function <vim/_editor.lua:0>
```
It's definitely related to https://github.com/neovim/neovim/commit/448907f65d6709fa234d8366053e33311a01bdb9 maybe there were other changes as well.

### Steps To Reproduce

1. Enable inlay hints
2. Using recent nightly neovim open a file will trigger LSP attach

### Expected Behavior

Inlay hints working properly.

### Repro

```Lua
-- DO NOT change the paths and don't remove the colorscheme
local root = vim.fn.fnamemodify("./.repro", ":p")

-- set stdpaths to use .repro
for _, name in ipairs({ "config", "data", "state", "cache" }) do
  vim.env[("XDG_%s_HOME"):format(name:upper())] = root .. "/" .. name
end

-- bootstrap lazy
local lazypath = root .. "/plugins/lazy.nvim"
if not vim.loop.fs_stat(lazypath) then
  vim.fn.system({ "git", "clone", "--filter=blob:none", "https://github.com/folke/lazy.nvim.git", lazypath, })
end
vim.opt.runtimepath:prepend(lazypath)

-- install plugins
local plugins = {
  "folke/tokyonight.nvim",
  "folke/LazyVim",
  -- add any other plugins here
}
require("lazy").setup(plugins, {
  root = root .. "/plugins",
})

vim.cmd.colorscheme("tokyonight")
-- add anything else here

-- sorry I don't know Lua well enough to provide minimal repro, made "plugin" like that:
-- lspconfig.lua
return {
  {
    "neovim/nvim-lspconfig",
    opts = {
      inlay_hints = { enabled = true },
    },
  },
}
```

**Comment by garymjr** [comment]
Since @folke is on vacation right now you can temporarily make the following change to `plugins/lsp/init.lua`
```diff
diff --git a/lua/lazyvim/plugins/lsp/init.lua b/lua/lazyvim/plugins/lsp/init.lua
index f0823e4..ed9c40f 100644
--- a/lua/lazyvim/plugins/lsp/init.lua
+++ b/lua/lazyvim/plugins/lsp/init.lua
@@ -120,7 +120,7 @@ return {
       if opts.inlay_hints.enabled and inlay_hint then
         Util.lsp.on_attach(function(client, buffer)
           if client.supports_method("textDocument/inlayHint") then
-            inlay_hint(buffer, true)
+            inlay_hint.enable(buffer, true)
           end
         end)
       end
```

**Comment by dpetka2001** [comment]
You can also add the following to your `keymaps.lua` to be able to toggle it on demand
```lua
if vim.lsp.inlay_hint then
  map("n", "<leader>uh", function() if vim.lsp.inlay_hint.is_enabled() then vim.lsp.inlay_hint.enable(0, false) else vim.lsp.inlay_hint.enable(0, true) end end, { desc = "Toggle Inlay Hints" })
end
```
until it gets fixed.

**Comment by jellydn** [comment]
Other option is rollback to the previous commit with tool like https://github.com/MordechaiHadad/bob

```sh
❯ bob install ad3568a70167ceb870931650afb7dcaed88640ec
❯ bob use ad3568a
❯ nvim --version

NVIM v0.10.0-dev-ad3568a
Build type: RelWithDebInfo
LuaJIT 2.1.1699524327
Run "nvim -V1 -v" for more info
```

**Comment by mati865** [comment]
I do use bob but I want to keep recent nightly for a different change. I have applied #2007 locally and it works good.

**Comment by grappas** [comment]
Same here

**Comment by mati865** [comment]
Fixed by #2007
